src/Service.Tests/SqlTests/GraphQLQueryTests/GraphQLQueryTestBase.cs

```diff
@@ -387,6 +387,209 @@ public virtual async Task MultipleResultJoinQuery()
+                {
+                  ""__typename"": ""StockConnection""
+                }
+            ";
+
+            JsonElement actual = await ExecuteGraphQLRequestAsync(graphQLQuery, graphQLQueryName, isAuthenticated: true);
+            SqlTestHelper.PerformTestEqualJsonStrings(expected, actual.ToString());
+        }
+
+        /// <summary>
+        /// Validates that a point query with only __typename field in the selection set
+        /// returns the right type
+        /// </summary>
+        [TestMethod]
+        public async Task PointQueryWithTypenameInSelectionSet()
+        {
+            string graphQLQueryName = "book_by_pk";
+            string graphQLQuery = @"{
+                book_by_pk(id: 3) {
+                    __typename
+                }
+            }";
+
+            string expected = @"
+                {
+                    ""__typename"": ""book""
+                }
+            ";
+
+            JsonElement actual = await ExecuteGraphQLRequestAsync(graphQLQuery, graphQLQueryName, isAuthenticated: false);
+            SqlTestHelper.PerformTestEqualJsonStrings(expected, actual.ToString());
+        }
+
+        /// <summary>
+        /// Validates that a nested point query with only __typename field in each selection set
+        /// returns the right types
+        /// </summary>
+        [TestMethod]
+        public async Task NestedPointQueryWithOnlyTypenameInEachSelectionSet()
+        {
+            string graphQLQueryName = "book_by_pk";
+            string graphQLQuery = @"{
+                book_by_pk(id: 3) {
+                    __typename
+                    publishers {
+                      __typename
+                      books {
+                        __typename
+                      }
+                    }
+                }     
+            }";
+
+            string expected = @"
+                {
+                  ""__typename"": ""book"",
+                  ""publishers"": {
+                    ""__typename"": ""Publisher"",
+                    ""books"": {
+                      ""__typename"": ""bookConnection""
+                    }
+                  }
+                }
+            ";
+
+            JsonElement actual = await ExecuteGraphQLRequestAsync(graphQLQuery, graphQLQueryName, isAuthenticated: false);
+            SqlTestHelper.PerformTestEqualJsonStrings(expected, actual.ToString());
+        }
+
+        /// <summary>
+        /// Validates that querying a SP with only __typename field in the selection set
+        /// returns the right type(s)
+        /// </summary>
+        public virtual async Task QueryAgainstSPWithOnlyTypenameInSelectionSet()
+        {
+            string graphQLQueryName = "executeGetBooks";
+            string graphQLQuery = @"{
+                executeGetBooks{
+                    __typename
+                }     
+            }";
+
+            string expected = @"
+                [
+      {
+        ""__typename"": ""GetBooks""
```

Comment:
Instead of hard coding this element "GetBooks" the required number of times, can't we generate such an expected string programmatically?

Reply:
Thanks for the suggestion! I have updated the test to dynamically generate this response string.